src/Search/SearchWithSuggestions/SearchWithSuggestions.js

```diff
@@ -223,9 +227,33 @@ class SearchWithSuggestions extends PureComponent {
         newCursor = cursor + 1
         break
       case 'Enter':
-        evt.target.blur()
-        selectedSuggestion = suggestions[cursor]
-        return selectedSuggestion && this.onSuggestionSelect(selectedSuggestion)
+        if (this.props.multipleCheck && this.state.searchTerm.includes(',')) {
+          const terms = this.state.searchTerm
+            .split(',')
+            .map(term => term.trim().toUpperCase())
+          const items = this.props.data[0].items
+          const foundItems = []
+          terms.forEach(term => {
+            const item = items.find(item => item.ticker === term)
+            if (item) foundItems.push(item)
+          })
+          const $this = this
+          foundItems.forEach(foundItem =>
+            $this.onSuggestionSelect({ item: foundItem })
+          )
+        } else {
+          selectedSuggestion = suggestions[cursor]
+          selectedSuggestion && this.onSuggestionSelect(selectedSuggestion)
+        }
+        this.setState({
+          suggestions: [],
+          searchTerm: undefined,
+          isFocused: false,
+          cursor: 0,
+          isSearching: false
+        })
+        document.dispatchEvent(new CustomEvent('clearsearchterminput'))
```

Comment:
It's hard to track such dispatched events, therefore in such cases callbacks are used. In our case, we need to access the `onClearClick` function from a `Search` component via ref and call it here.